### PR TITLE
adding finished Kata files

### PR DESCRIPTION
**Pull request by ian-antking**

### CODE REVIEW

File: __tests__/kata1.fizzBuzz.test.js [hunk 1]
```diff
@@ -1,19 +1,22 @@
-const { fizzBuzz } = require('../src');
+const fizzBuzz = require('../src/kata1.fizzBuzz.js')
 
 describe('fizzBuzz', () => {
   it('returns Fizz when passed a multiple of 3', () => {
-
+    expect(fizzBuzz(3)).toBe('Fizz')
   });
```
Comment (ian-antking) [inline __tests__/kata1.fizzBuzz.test.js:7]: It might be a good idea to add a few more expect lines to these tests. You could check other multiples of 3 and 5.

File: __tests__/kata2.booleanToWord.test.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-const { booleanToWord } = require('../src');
+const booleanToWord = require('../src/kata2.booleanToWord.js')
 
 describe('booleanToWord', () => {
-  // how do we create specs again???
+  it ('returns true when passed yes', () => { 
+    expect(booleanToWord('yes')).toBe(true)
```
Comment (ian-antking) [inline __tests__/kata2.booleanToWord.test.js:5]: Does this also work if you pass it `'no'`?

File: __tests__/kata3.numberToReversedDigits.test.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-const { numberToReversedDigits } = require('../src');
+const numberToReversedDigits = require('../src/kata3.numberToReversedDigits.js')
 
 describe('numberToReversedDigits', () => {
   it('returns a reversed array of the number\'s digits', () => {
-    
+    expect(numberToReversedDigits(12345)).toEqual(['5', '4', '3', '2', '1'])
```
Comment (ian-antking) [inline __tests__/kata3.numberToReversedDigits.test.js:5]: What would happen if this was passed an array of numbers as strings?

`['1', '2', '3', '4', '5']`

File: __tests__/kata4.humanCatDogYears.test.js [hunk 4]
```diff
@@ -1,3 +1,9 @@
-const { humanCatDogYears } = require('../src');
+const humanCatDogYears = require('../src/kata4.humanCatDogYears.js');
 
-// Look Ma, no handlebars!!!
+describe('humanCatDogYears', () => {
+    it('returns an array of human years cat years and dog years', () => {
+      expect(humanCatDogYears(12)).toEqual([12, 64, 74])
+      expect(humanCatDogYears(4)).toEqual([4, 32, 34])
+      expect(humanCatDogYears(7)).toEqual([7, 44, 49])
```
Comment (ian-antking) [inline __tests__/kata4.humanCatDogYears.test.js:7]: Good work adding a range of test cases. It's also important to test for "edge cases". What would happen if you passed this function a 1, 0 or even a negative number?

File: __tests__/kata7.getEmployerRole.test.js [hunk 5]
```diff
@@ -1,7 +1,19 @@
...
+}, {
+  name: 'Javid',
+  role: 'Human Recommended Reading Assistant'
+}]
 
 describe('getEmployerRole', () => {
   it('returns the employee\'s role in the company', () => {
-    
+    const employeeName = 'Jenny'
+    const employeeRole = getEmployerRole(employeeName, employees)
```
Comment (ian-antking) [inline __tests__/kata7.getEmployerRole.test.js:16]: Is this line necessary? It doesn't look like you're doing anything with the `const` after you define it.

File: src/kata1.fizzBuzz.js [hunk 6]
```diff
@@ -1,5 +1,17 @@
 const fizzBuzz = (number) => {
-
+    const isDivisableBy3 = (number % 3) === 0
+    const isDivisableBy5 = (number % 5) === 0
+
```
Comment (ian-antking) [inline src/kata1.fizzBuzz.js:4]: I like that you separated this logic from your `if` statements!

File: src/kata2.booleanToWord.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
-const booleanToWord = (boolean) => {
-
-}
+const booleanToWord = (word) => {
```
Comment (ian-antking) [inline src/kata2.booleanToWord.js:1]: I don't think this would work if I passed it a 'no'. Try adding another test and trying to pass it.

File: src/kata3.numberToReversedDigits.js [hunk 8]
```diff
@@ -1,5 +1,10 @@
 const numberToReversedDigits = (number) => {
-
-}
-
+    number = 12345
+    const initialArray = ('' + number).split('');
+    const reversedArray = [];
+    for (let i = initialArray.length - 1; i >= 0; i--) {
```
Comment (ian-antking) [inline src/kata3.numberToReversedDigits.js:5]: A for loop works just fine here, but you can really cut down on the amount of code if you chain a `.reverse()` onto your `.split('')`.

File: src/kata4.humanCatDogYears.js [hunk 9]
```diff
@@ -1,5 +1,16 @@
 const humanCatDogYears = (number) => {
+    
+    let firstCatYear = 15;
+    let secondCatYear = 9;
+    let laterCatYears = ((number - 2) * 4);
+    let firstDogYear = 15;
+    let secondDogYear = 9;
+    let laterDogYears = ((number - 2) * 5);
+
```
Comment (ian-antking) [inline src/kata4.humanCatDogYears.js:9]: A couple of things here:
- It's generally better to use `const` over `let`, as it prevents our variables changing into things we don't expect. Especially here as nothing is being re-assigned later.

- If we are using variables as a constant value, we tend to write them in all caps to let other developers know.

- I don't think this code would work if it were passed a 0, 1 or two. You need to add some logic to handle those situations.

File: src/kata5.reachDestination.js [hunk 10]
```diff
@@ -1,5 +1,7 @@
 const reachDestination = (distance, speed) => {
-
+    let estimatedArrivalTime = Math.round((distance / speed)*2)/2;
+    
+    return (`I will arrive in ${estimatedArrivalTime} hours`);
```
Comment (ian-antking) [inline src/kata5.reachDestination.js:4]: Good work, you don't need the `()` around your return string though.

File: src/kata6.joinNames.js [hunk 11]
```diff
@@ -1,5 +1,14 @@
 const joinNames = (namesObj) => {
 
+    let names = [];
+
+    for (let i = 0; i < namesObj.length; i++) {
+    names.push(namesObj[i])
+    };
+
```
Comment (ian-antking) [inline src/kata6.joinNames.js:8]: The logic here is great! Just remember to use const rather than let.

Comment (ian-antking) [reply src/kata6.joinNames.js:8]: Unless it's something like a for loop, where you need to use `let`.

**Review by ian-antking — COMMENTED** [review]
Solid work. I've added some comments and suggestions. Mainly there are a few tests which could use a few more `expects` adding to, and then a bit more code to handle extreme cases.

Keep it up!
